src/utxo/repo.rs

```diff
@@ -445,4 +445,22 @@ impl UtxoRepo {
 
         Ok(usize::try_from(res.average).expect("Could convert to usize"))
     }
+
+    pub async fn find_delete_utxo(
+        &self,
+        tx: &mut Transaction<'_, Postgres>,
+        outpoint: bitcoin::OutPoint,
+    ) -> Result<Option<LedgerTransactionId>, UtxoError> {
+        let row = sqlx::query!(
+            r#"DELETE FROM bria_utxos 
+            WHERE tx_id = $1 AND vout = $2
```

Comment:
must qualify `keychain_id` as well.
Also if it is already settled (`income_settled_ledger_tx_id IS NOT NULL`) then we must return an error and be alerted.